FinSearchUnified/Tests/Bundle/StoreFrontBundle/Gateway/Findologic/CustomFacetGatewayTest.php

```diff
@@ -0,0 +1,416 @@
+            ['ShopKey', 'ABCD0815'],
+            ['ActivateFindologicForCategoryPages', false],
+            ['IntegrationType', Constants::INTEGRATION_TYPE_API]
+        ];
+
+        // Create mock object for Shopware Config and explicitly return the values
+        $mockConfig = $this->getMockBuilder('\Shopware_Components_Config')
+            ->setMethods(['offsetGet'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $mockConfig->method('offsetGet')
+            ->willReturnMap($configArray);
+
+        // Assign mocked config variable to application container
+        Shopware()->Container()->set('config', $mockConfig);
+
+        $sessionArray = [
+            ['isSearchPage', true],
+            ['isCategoryPage', false],
+            ['findologicDI', false]
+        ];
+
+        // Create mock object for Shopware Session and explicitly return the values
+        $mockSession = $this->getMockBuilder('\Enlight_Components_Session_Namespace')
+            ->setMethods(['offsetGet'])
+            ->getMock();
+        $mockSession->method('offsetGet')
+            ->willReturnMap($sessionArray);
+
+        // Assign mocked session variable to application container
+        Shopware()->Container()->set('session', $mockSession);
+
+        $mockOriginalService = $this->getMockBuilder(
+            '\Shopware\Bundle\StoreFrontBundle\Gateway\DBAL\CustomFacetGateway'
+        )
+            ->setMethods(['getList'])
+            ->disableOriginalConstructor()
+            ->getMock();
+        $mockOriginalService->expects($this->never())->method('getList');
+
+        $mockUrlBuilder = $this->getMockBuilder('\FinSearchUnified\Helper\UrlBuilder')->getMock();
+        $mockUrlBuilder->expects($this->once())->method('setCustomerGroup');
+        $xmlResponse = new SimpleXMLElement('<?xml version="1.0" encoding="UTF-8"?><searchResult></searchResult>');
+
+        $results = $xmlResponse->addChild('results');
+        $results->addChild('count', 2);
+        $filters = $xmlResponse->addChild('filters');
+        foreach ($filterData as $data) {
+            $filter = $filters->addChild('filter');
+            foreach ($data as $key => $value) {
+                $filter->addChild($key, $value);
+            }
+        }
+        $mockUrlBuilder->expects($this->once())
+            ->method('buildCompleteFilterList')
+            ->willReturn(new \Zend_Http_Response(200, [], $xmlResponse->asXML()));
+
+        $originalHydrator = Shopware()->Container()->get('fin_search_unified.custom_listing_hydrator');
+
+        $mockHydrator = $this->getMockBuilder(
+            '\FinSearchUnified\Bundle\StoreFrontBundle\Gateway\Findologic\Hydrator\CustomListingHydrator'
+        )
+            ->setMethods([])
+            ->getMock();
+        $facetArray = [];
+        foreach ($xmlResponse->filters->filter as $filter) {
+            $facetArray[] = $originalHydrator->hydrateFacet($filter);
```

Comment:
Since you have to call `hydrateFacet` anyway, why not drop the mock and directly use `CustomListingHydrator`?